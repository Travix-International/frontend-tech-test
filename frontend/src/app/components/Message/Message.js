import React from 'react';
import PropTypes from 'prop-types';
import Section from '../Section';
import { Wrapper, Title, Description } from './assets/style';

const Message = ({ title, description, type }) => (
  <Section>
    <Wrapper type={type}>
      <Title>{title}</Title>
      <Description>{description}</Description>
    </Wrapper>
  </Section>
);

Message.propTypes = {
  type: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export default Message;
