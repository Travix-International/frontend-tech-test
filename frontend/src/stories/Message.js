import React from 'react';
import { storiesOf } from '@storybook/react';
import Message from '../app/components/Message';

storiesOf('Message', module)
  .add('Success', () => <Message type="success" title="Success" description="Thank you for testing my stories." />)
  .add('Error', () => <Message type="error" title="Error" description="I thought it make you happy." />)
  .add('Warning', () => <Message type="warning" title="Warning" description="Install ubuntu to support open source." />)
  .add('Info', () => (
    <Message type="info" title="Information" description="Look around! there are many resources to learn out there." />
  ));
