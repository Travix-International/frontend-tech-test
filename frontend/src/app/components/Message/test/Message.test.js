import React from 'react';
import { shallow } from 'enzyme';
import Message from '../Message';

it('Message renders correctly', () => {
  const MessageSnapshot = shallow(<Message title="Error" description="Something wrong.." type="error" />);
  expect(MessageSnapshot).toMatchSnapshot();
});
