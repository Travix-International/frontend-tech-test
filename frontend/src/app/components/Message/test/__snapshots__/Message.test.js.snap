// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message renders correctly 1`] = `ShallowWrapper {}`;
